src/privatesend/privatesend-client.cpp

```diff
@@ -1359,6 +1355,10 @@ bool CPrivateSendClientSession::PrepareDenominate(int nMinRounds, int nMaxRounds
         return false;
     }
 
+    for (const auto& pair : vecPSInOutPairsRet) {
```

Comment:
Shouldn't `vpwallets[0]->LockCoin()` also be called for these?

Reply:
We lock all coins above https://github.com/dashpay/dash/pull/3194/files#diff-5b62dcd1a550f736f5b375f7623896baR1297 and later unlock only unused ones (e.g. https://github.com/dashpay/dash/pull/3194/files#diff-5b62dcd1a550f736f5b375f7623896baR1312) or all coins if smth went wrong (e.g. https://github.com/dashpay/dash/pull/3194/files#diff-5b62dcd1a550f736f5b375f7623896baR1351). Doing so helps to prevent Here we just store which coins were kept locked after al the calculations to be able to unlock them after mixing is done (by either calling `UnlockCoins()` directly or via `ResetPool()`). But that's kind of counterintuitive, I see now...

Reply:
New code is much better :+1: